sled-diagnostics/src/logs.rs

```diff
@@ -501,13 +501,24 @@ impl LogsHandle {
                         .filter(|f| is_log_file(f.path(), filename))
                     {
                         let logfile = f.path();
+                        let system_mtime =
+                            f.metadata().and_then(|m| m.modified()).inspect_err(|e| {
```

Comment:
`metadata()` here doesn't follow symlinks...which I believe is okay since we are operating from the gz context.

Reply:
Right, this is fine today. On the one hand I don't foresee us copying over symlinks from a sled zip, but there's no harm in handling them just in case.  Switched to `Utf8Path::metadata` which will traverse symlinks in fa07354f2fd7b16310611423e5a1fbd6d117c9eb.